tabular/src/autogluon/tabular/models/widedeep_nn/widedeep_nn_model.py

```diff
@@ -0,0 +1,289 @@
+import logging
+import time
+from typing import Union
+
+import numpy as np
+import torch
+
+from autogluon.common.features.types import R_INT, R_FLOAT, R_DATETIME, R_BOOL, R_CATEGORY
+from autogluon.common.features.types import R_OBJECT, S_TEXT_NGRAM, S_TEXT_AS_CATEGORY
+from autogluon.core import metrics
+from autogluon.core.constants import BINARY, REGRESSION, MULTICLASS
+from autogluon.core.models import AbstractModel
+from autogluon.core.utils.exceptions import TimeLimitExceeded
+from autogluon.core.utils.files import make_temp_directory
+from autogluon.core.utils.try_import import try_import_pytorch_widedeep
+from .hyperparameters.parameters import get_param_baseline
+from .hyperparameters.searchspaces import get_default_searchspace
+from .metrics import get_nn_metric, get_objective, get_monitor_metric
+from .preprocessing_utils import MissingFiller, TargetScaler, CategoricalFeaturesFilter
+from .utils import set_seed
+
+logger = logging.getLogger(__name__)
+
+
+class WideDeepNNModel(AbstractModel):
+
+    def __init__(self, path: str = None, name: str = None, problem_type: str = None, eval_metric: Union[str, metrics.Scorer] = None, hyperparameters=None):
+        super().__init__(path, name, problem_type, eval_metric, hyperparameters)
```

Comment:
just do `def __init__(self, **kwargs):`

This makes life easier.

Reply:
updated